asyncio_mqtt/client.py

```diff
@@ -312,20 +327,37 @@ async def force_disconnect(self) -> None:
             self._disconnected.set_result(None)
 
     @_outgoing_call
-    async def subscribe(self, *args: Any, timeout: int = 10, **kwargs: Any) -> int:
-        result, mid = self._client.subscribe(*args, **kwargs)
+    async def subscribe(
+        self,
+        topic: Union[
+            str,
+            tuple[str, mqtt.SubscribeOptions],
+            list[tuple[str, mqtt.SubscribeOptions]],
+            list[tuple[str, int]],
+        ],
+        qos: int = 0,
+        options: Optional[mqtt.SubscribeOptions] = None,
+        properties: Optional[Properties] = None,
+        timeout: int = 10,
+    ) -> int:
```

Comment:
I like the idea to explicitly type these parameters. Makes it easier to use the library.

I know that you exhaust the current argument list for `paho.Client.subscribe`. In any case, let's keep the `*args` and `**kwargs` forwarding just for the sake of future compatibility. 👍 This way, if paho adds a new argument, asyncio-mqtt is still compatible (even though we don't expose explicit typings for ssaid argument).

Of course, this goes for the other functions as well.